### PR TITLE
Set knowhere thread pool size in indexnode

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/milvus-io/milvus/issues/24177

### CODE REVIEW

File: internal/indexnode/indexnode.go [hunk 1]
```diff
@@ -162,6 +163,9 @@ func (i *IndexNode) initSegcore() {
 	cCPUNum := C.int(hardware.GetCPUNum())
 	C.InitCpuNum(cCPUNum)
 
+	cKnowhereThreadPoolSize := C.uint32_t(hardware.GetCPUNum())
```
Comment (xige-16) [inline internal/indexnode/indexnode.go:166]: It wouldbe better to add a pool size ratio to milvus.yaml

**Review by congqixia — APPROVED** [review]
after offline discussion, we shall add a configuration item with upper limit to 1 in following prs
@cqy123456 
/lgtm

**Pull request merged**